Deprecate `Benchmark.ms` and add `benchmark` to the gemspec

Earlopain opened the pull request:
### Motivation / Background

Ruby plans to make `benchmark` a bundled gem: https://github.com/ruby/ruby/pull/11492

It doesn't make much sense to keep this core extension since rails would need to add it to its gemspec to continue offering it for what basically amounts to `1000 * x`.

Since requiring it could emit a warning, add it to the gemspec until it can later be removed again with Rails 8.1

The benchmark generator will continue to work as usual: It adds `benchmark-ips` to the users Gemfile, which in turn should depend on `benchmark`: https://github.com/evanphx/benchmark-ips/pull/141

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

rafaelfranca reviewed (changes requested):
Duplicating this code everywhere isn't no go. I get not wanting to add the dependency, but instead of all this duplication, I'd just inline `Benchmark.realtime` and `ms` in Rails.

Earlopain commented:
Will keep an eye on the ruby PR and do that when/if it gets merged. I was premature and now there is a bit of discussion happening there.

Earlopain commented:
Looks like it will become bundled, PR has been merged at https://github.com/ruby/ruby/pull/11560. I added `ActiveSupport::Benchmark` which should address your feedback. I also openend https://github.com/rails/rails/pull/52815 for 7-2.

bogdan commented on activesupport/lib/active_support/benchmark.rb:
```diff
@@ -0,0 +1,27 @@
+    #
+    #   ActiveSupport::Benchmark.realtime { User.expensive_query.count }
+    #   # => 5.8273
+    def self.realtime(&block)
```
Is there a need for this method? `::Benchmark.realtime` is exactly the same: https://github.com/ruby/benchmark/blob/master/lib/benchmark.rb#L311.

Earlopain replied:
The whole point of this PR is that `benchmark` will need to be explicitly be added to the dependencies of rails (because ruby will bundle it). I'm of the oppinion that for what amounts to 3 lines of code is not warranted.

Of course it needs to be added to dependencies right now anyways to suppress the warning, but this PR allows to remove it in future versions.

bogdan commented on activesupport/test/benchmark_test.rb:
```diff
@@ -0,0 +1,15 @@
+  end
+
+  def test_ms
+    ms = ActiveSupport::Benchmark.realtime { sleep 0.1 }
```
```suggestion
    ms = ActiveSupport::Benchmark.ms { sleep 0.1 }
```

Earlopain replied:
🤦, thanks

bogdan commented on activesupport/lib/active_support/benchmark.rb:
```diff
@@ -0,0 +1,27 @@
+    #   ActiveSupport::Benchmark.ms { User.all }
+    #   # => 0.074
+    def self.ms(&block)
+      1000 * realtime(&block)
```
It would be better if a method would be implemented in https://github.com/ruby/benchmark. Do you think we should try a PR?

byroot commented:
IMO the value of this `Benchmark.ms` core ext is really minimal. I'd rather deprecate it without replacement, and I doubt much people are using it.

As for the internal usages, I only see two, so:

```ruby
start = Process.clock_gettime(Process::CLOCK_MONOTONIC, :float_millisecond)
do_thing
duration_ms = Process.clock_gettime(Process::CLOCK_MONOTONIC, :float_millisecond) - start
```

Earlopain commented:
> IMO the value of this Benchmark.ms core ext is really minimal. I'd rather deprecate it without replacement, and I doubt much people are using it.

I would agree.

> As for the internal usages, I only see two, so

That's pretty neat, TIL. I think accepting a unit instead for `realtime` is the nicer solution, it barely adds any code.
I've updated the `Benchmark.ms` deprecation to make note of that, but that depends on the notion that this method should even be exposed to users. Let me know if I should mark it `nodoc` instead.

rafaelfranca commented on activesupport/lib/active_support/benchmark.rb:
```diff
@@ -0,0 +1,22 @@
+
+module ActiveSupport
+  # = \Benchmark
+  module Benchmark
```
This should be `# :nodoc:` I don't want people using it.

Earlopain replied:
Yeah, sure. Updated deprecation message/changelog

Earlopain committed:
Deprecate `Benchmark.ms` and add `benchmark` to the gemspec

Ruby plans to make `benchmark` a bundled gem: https://github.com/ruby/ruby/pull/11492

It doesn't make much sense to continue this core extension since rails would
need to add it to its gemspec to continue offering it for what basically amounts to `x * 1000`.

Since requiring it could emit a warning, add it to the gemspec until it can later be removed
again with Rails 8.1

The benchmark generator will continue to work as usual:
It adds `benchmark-ips` to the users Gemfile.

ghiculescu commented:
> I doubt much people are using it.

I use it all the time :( I guess I'll be re-implementing it in an intializer.

chaadow commented on activesupport/lib/active_support/benchmark.rb:
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module ActiveSupport
+  module Benchmark # :nodoc:
```
hey @Earlopain i can do the PR, but is it possible to remove this `:nodoc` because there is actually a doc right below :D 

and i couldnt find it here https://edgeapi.rubyonrails.org/files/activesupport/lib/active_support/core_ext/benchmark_rb.html

thanks!
